WcaOnRails/app/models/competition.rb

```diff
@@ -21,12 +21,15 @@ class Competition < ActiveRecord::Base
   MAX_ID_LENGTH = 32
   MAX_NAME_LENGTH = 50
   validates :id, presence: true, uniqueness: true, length: { maximum: MAX_ID_LENGTH },
-                 format: { with: /\A[a-zA-Z0-9]+\Z/ }
+                 format: { with: /\A[a-zA-Z0-9]+\Z/ }, if: :name_valid_or_updating?
+  private def name_valid_or_updating?
+    self.persisted? ? true : name.length <= MAX_NAME_LENGTH && name =~ ENDS_WITH_YEAR_RE
```

Comment:
I want to perform this validation all the time on the edit page. Because someone could update the name and the cellName on the validation page and at that point I want to display both errors to the user. It's only on the new page that I don't want to validate the id and cellName if the name itself isn't valid.